mathesar/utils/preview.py

```diff
@@ -1,27 +1,52 @@
 import re
+from collections import defaultdict, namedtuple
 
-from db.constraints.utils import ConstraintType
-from mathesar.models.base import Column, Constraint
+from db.metadata import get_empty_metadata
+from db.tables.operations.select import get_joinable_tables
+from mathesar.models.base import Column, Table
 
 
-def _preview_info_by_column_id(fk_constraints, previous_path=[], exising_columns=[]):
+def _preview_info_by_column_id(
+    referrer_table,
+    referent_column_obj_by_column_attnum,
+    prefetched_objects,
+    previous_path=[],
+    exising_columns=[]
+):
     preview_info = {}
     preview_columns = exising_columns
-    for fk_constraint in fk_constraints:
-        constrained_column = fk_constraint.columns[0]
-        # For now only single column foreign key is used.
-        referent_column = fk_constraint.referent_columns[0]
-        referent_table = referent_column.table
+    for fk_constrained_column_attnum, referent_tuple in referent_column_obj_by_column_attnum.items():
+        constrained_column = next(
+            constrained_column
+            for constrained_column in prefetched_objects.possible_columns
+            if fk_constrained_column_attnum == constrained_column.attnum and constrained_column.table.oid == referrer_table.oid
+        )
```

Comment:
I thing this pattern adds a bunch of noise and pain to the function. Wouldn't it be possible to make it so that `prefetched_objects.possible_columns` is a dictionary (or some other mapping type)? Then you could just get the one you want from it.